springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/components/DefaultComponentsService.java

```diff
@@ -145,10 +145,25 @@ private String getSchemaName(Class<?> type, Map<String, Schema> schemas) {
     }
 
     private void preProcessSchemas(Map<String, Schema> schemas, String schemaName, Class<?> type) {
+        processCommonModelConverters(schemas);
         processAsyncApiPayloadAnnotation(schemas, schemaName, type);
         processSchemaAnnotation(schemas, schemaName, type);
     }
 
+    private void processCommonModelConverters(Map<String, Schema> schemas) {
+        schemas.values().stream()
+                .filter(schema -> schema.getType() == null)
+                .filter(schema -> schema.get$ref() != null)
+                .forEach(schema -> {
+                    String targetSchemaName = schema.getName();
+                    String sourceSchemaName = StringUtils.substringAfterLast(schema.get$ref(), "/");
+
+                    Schema<?> actualSchema = schemas.get(sourceSchemaName);
```

Comment:
Line 159 and 161 seem to belong together as the are actually resolving a schema from a ref. I am wondering why you split them up?